app/src/main/java/com/lastaoutdoor/lasta/data/db/UserDBRepositoryImpl.kt

```diff
@@ -124,6 +124,53 @@ class UserDBRepositoryImpl @Inject constructor(context: Context, database: Fireb
     userDocumentRef.update("favorites", FieldValue.arrayUnion(activityId)).await()
   }
 
+  override suspend fun getUsersByUsernameWithSubstring(query: String): List<UserModel> {
+    // if the query is empty, return an empty list
+    if (query.isEmpty() || query.isBlank()) {
+      return emptyList()
+    }
+    // get all users
+    val allUsersUsernames = userCollection.get().await()
+    if (allUsersUsernames.isEmpty) {
+      return emptyList()
+    }
+    val usersToList = mutableListOf<UserModel>()
+    // for each user, check if the username contains the query
+    allUsersUsernames.documents.forEach { doc ->
+      // get the username of the user
+      val userName = doc.getString("userName") ?: ""
+      // if the username contains the query, return the user
+      if (userName.lowercase().contains(query)) {
+        usersToList.add(
+            UserModel(
+                userId = doc.id,
```

Comment:
This conversion of document firebase to UserModel is nice but it deserves a method of it's own I believe. Check the methods from `UserModel.kt` and `Converters.kt` inside the package models/user to check if there exists already anything useful, and if it doesn't, then add the following method that could be reused then after by someone else!

Reply:
Yep changed it!